inc/zoo/swar/SWAR.h

```diff
@@ -472,13 +481,53 @@ booleans(SWAR<NB, T> arg) noexcept {
 template<int NBits, typename T>
 constexpr auto
 differents(SWAR<NBits, T> a1, SWAR<NBits, T> a2) {
-    return booleans(a1 ^ a2);
+    return ~equals(a1, a2);
 }
 
+/**
+ * @return BooleanSWAR that contains a true value in each lane where the two
+ * inputs SWARs match.
+ */ 
+template<int NBits, typename T>
+constexpr BooleanSWAR<NBits, T>
+equals(SWAR<NBits, T> a1, SWAR<NBits, T> a2) noexcept {
+    // Knuth, TAOCP 4A pg 152
+    using S = swar::SWAR<NBits, T>;
+    const S x = (a1 ^ a2);
+    return BooleanSWAR<NBits, T>{S{S::MostSignificantBit} & ~(x| ((x|S{S::MostSignificantBit}) - S{S::LeastSignificantBit}))};
```

Comment:
```suggestion
    constexpr auto
        TAOCP_H{S{S::MostSignificantBit}},
        TAOCP_L{S{S::LeastSignificantBit}};
    auto
           nullLaneIfEqual = a1 ^ a2,
           allowSubtractionByOneTurningOnMSB = nullLaneIfEqual | TAOCP_H,
           nullDetectorViaFlippingMSB_low = allowSubtractionByOneTurningOnMSB - TAOCP_L,
           equalLanesHaveMSB_off = x | nullDetectorViaFlippingMSB_low,
           cheapFlippingOfMSB = ~equalLanesHaveMSB_off,
           rv = TAOCP_H & cheapFlippingOfMSB;
    
    return {rv};
```

Reply:
I thought the opinion was "if we don't really know why it works explaining it isn't really useful"?

Reply:
Might it also be useful to (have a version that) elides that final & sometimes?  The msb are still on, and we might be doing a later bit op with some other booleanSWAR anyway....

Reply:
I have gotten to understand this, if you see, the names explain what is going on.  If this is not understood, then I failed, but it can be complemented with a comment.  Try to understand the explanation and let me know what is missing.

Reply:
'allowSubtraction' is a little indistinct, since "ensure MSB on so subtracts won't roll other lanes" is a bit wordy. I don't have a better candidate though. 
used 'x' instead of nullLaneIfEqual
init list return failed on compile.
Looks fine enough.